hudi-utilities/src/test/java/org/apache/hudi/utilities/deltastreamer/TestHoodieDeltaStreamer.java

```diff
@@ -1395,19 +1403,26 @@ public void testAsyncClusteringServiceWithConflicts(HoodieRecordType recordType)
   }
 
   @ParameterizedTest
-  @EnumSource(value = HoodieRecordType.class, names = {"AVRO", "SPARK"})
+  @EnumSource(value = HoodieRecordType.class, names = {"AVRO","SPARK"})
   public void testAsyncClusteringServiceWithCompaction(HoodieRecordType recordType) throws Exception {
-    String tableBasePath = basePath + "/asyncClusteringCompaction";
+    String tableBasePath = basePath + "/asyncClusteringCompaction" + RANDOM.nextInt(1000);
     // Keep it higher than batch-size to test continuous mode
     int totalRecords = 2000;
 
-    // Initial bulk insert
-    HoodieDeltaStreamer.Config cfg = TestHelpers.makeConfig(tableBasePath, WriteOperationType.INSERT);
+    // user upsert as operation type.
+    HoodieDeltaStreamer.Config cfg = TestHelpers.makeConfig(tableBasePath, WriteOperationType.UPSERT);
```

Comment:
Will async clustering fail due to the conflict between clustering and updates in the same file group?

Reply:
Here we are testing that async clustering can run in concurrently for an MOR table while async compaction is enabled. So, may be we can enable allow update strategy. From my runs, as soon as replace commit succeeds, the test ends. probably I did not hit it locally in few runs I had.

Reply:
yeah i think we need to curate carefully such a scenario. SparkAllowUpdaterStrategy does not really resolve the write-write conflict. It will simply rollback clustering if `ROLLBACK_PENDING_CLUSTERING_ON_CONFLICT` is configured as well.